packages/interface/src/hooks/useRegistry.ts

```diff
@@ -0,0 +1,50 @@
+import { DefaultError, useMutation, UseMutationResult } from "@tanstack/react-query";
+import { useAccount } from "wagmi";
+
+import { approveApplication, submitApplication } from "~/utils/registry";
+
+import type { Hex, TransactionReceipt } from "viem";
+
+/**
+ * Submit an application for approval
+ *
+ * @returns whether the submission was successful
+ */
+export function useSubmitApplication(): UseMutationResult<
+  TransactionReceipt,
+  DefaultError,
+  { refUID?: string | undefined; metadataUrl: string; registryAddress: Hex; recipient: `0x${string}` }
```

Comment:
And let's move it to an interface and use below for `useMutation`
```suggestion
  { refUID?: string; metadataUrl: string; registryAddress: Hex; recipient: Hex }
```